### PR TITLE
Add FXIOS-9928 Bundle DefaultSuggestedSites and Workaround for Blurry Google Favicons [Favicon Refactor]

### PR DESCRIPTION
**Pull request by ih-codes**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-9928)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/21783)

## :bulb: Description
This PR bundles high quality favicon assets for the `DefaultSuggestedSites` and the Google pinned tile.

As well, this PR introduces a workaround for all Google websites to display the bundled G icon.

## Demo
**A. First launch:**
**Note**: The `TopSiteCell`s reload 3 times on fresh launch, so there’s still a flicker, but the images pop in immediately for even slow connections since they’re bundled.

https://github.com/user-attachments/assets/933df295-f801-43fa-8cb3-56b7d742b5b9

**B. Google tiles on Home** (pinned Google tile manually removed so Maps would appear...): 
Screenshot 2024-09-05 at 8 05 28 PM

### Notes

In order to use BrowserKit's SiteImageView's exported types in the static library Storage, we had to add `SiteImageView` to each of Storage, Sync, and ClientTests targets under Build Phases > Link Binary With Libraries. As well, it was necessary to add a second copy of Kingfisher to the Client's SPM configuration (strange). Then we also had to link with Kingfisher in the same places we added SiteImageView.

Without doing this, we would get some very cryptic linker errors, and not even in the target we were working on, as well as flagging "CoreAudioTypes" as the issue (red herring).

Screenshot 2024-09-05 at 12 44 58 PM

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [x] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: BrowserKit/Sources/SiteImageView/ImageProcessing/ImageHandler.swift [hunk 1]
```diff
@@ -114,4 +126,11 @@ class DefaultImageHandler: ImageHandler {
             return UIImage(named: "globeLarge")?.withRenderingMode(.alwaysTemplate) ?? UIImage()
         }
     }
+
+    private func getBundleImage(assetName: String) throws -> UIImage {
+        guard let image = UIImage(named: assetName) else {
+            throw SiteImageError.noImageInBundle
```
[the remark this answers is not in the document]
Comment (OrlaM) [reply BrowserKit/Sources/SiteImageView/ImageProcessing/ImageHandler.swift:132]: You can log from here without going through the client. Logger is a singleton that lives in Common. See DefaultSiteImageDownloader as an example.

**Review by OrlaM — APPROVED** [review]
LGTM

File: BrowserKit/Sources/SiteImageView/ImageProcessing/ImageHandler.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (ih-codes) [reply BrowserKit/Sources/SiteImageView/ImageProcessing/ImageHandler.swift:143]: Oh perfect, thanks! 👍

**Comment by ih-codes** [comment]
@Mergifyio backport release/v131

**Pull request merged**